protocol/bc/transaction.go

```diff
@@ -19,35 +18,43 @@ const CurrentTransactionVersion = 1
 // Tx holds a transaction along with its hash.
 type Tx struct {
 	TxData
-	TxHashes `json:"-"`
+	*TxEntries `json:"-"`
 }
 
 func (tx *Tx) UnmarshalText(p []byte) error {
 	if err := tx.TxData.UnmarshalText(p); err != nil {
 		return err
 	}
 
-	hashes, err := ComputeTxHashes(&tx.TxData)
+	txEntries, err := MapTx(&tx.TxData)
 	if err != nil {
 		return errors.Wrap(err)
 	}
 
-	tx.TxHashes = *hashes
+	tx.TxEntries = txEntries
 	return nil
 }
 
+func (tx *Tx) IssuanceHash(n int) Hash {
+	return tx.TxEntries.TxInputIDs[n]
+}
+
+func (tx *Tx) OutputID(outputIndex uint32) Hash {
```

Comment:
Same recommendation here as for IssuanceHash, this param should be int. (Oh I see it was uint32 before. This isn't a new method, it just moved up here. But now seems like an ok time to fix it.)